src/corehost/cli/bdl_file_entry.h

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef __BDL_FILE_ENTRY_H__
+#define __BDL_FILE_ENTRY_H__
+
+#if FEATURE_APPHOST
+
+#include <cstdint>
+#include "bdl_file_type.h"
+#include "pal.h"
+
+#pragma pack(push, 1)
+
+// FileEntry: Records information about embedded files.
+// 
+// The bundle manifest records the following meta-data for each 
+// file embedded in the bundle:
+// Fixed size portion (represented by file_entry_inner_t)
+//   - Offset     
+//   - Size       
+//   - File Entry Type       
+//   - path-length  (7-bit extension encoding, 1 Byte due to MAX_PATH)
+// Variable Size portion
+//   - relative path  ("path-length" Bytes)
+
+class file_entry_t
+{
+public:
+    
+    // The inner structure represents the fields that can be 
+    // read contiguously for every file_entry. 
+    struct file_entry_inner_t
+    {
+        int64_t offset;
+        int64_t size;
+        file_type_t type;
+        int8_t path_length;
+    } data;
+    pal::string_t relative_path; // Path of an embedded file, relative to the extraction directory.
```

Comment:
Naming: most classes in the host use the `m_` prefix for member fields. Even in this change the processor uses this, but the file entry here doesn't.

Reply:
My understanding is that only *private* data members are prefixed by `m_` and *private* static members are prefixed by `s_` ([example](https://github.com/dotnet/core-setup/blob/master/src/corehost/cli/fx_definition.h#L36)). Public data members don't have a prefix, and they read like properties ([example](https://github.com/dotnet/core-setup/blob/master/src/corehost/cli/host_startup_info.h#L28)).

The convention used in other header files in this repo is not consistent, but I haven't seen any use of `m_` prefix for public members in the code-base. If you think its better, I can make the fields private and expose accessor functions.

Reply:
I missed the fact that these are public...
OK - I guess I'm fine with it... still not a fan, but it's OK.